fft/src/KokkosFFT_Transform.hpp

```diff
@@ -123,9 +123,10 @@ namespace KokkosFFT {
 /// \param exec_space [in] Kokkos execution space
 /// \param in [in] Input data (complex)
 /// \param out [out] Ouput data (complex)
-/// \param norm [in] How the normalization is applied (optional)
-/// \param axis [in] Axis over which FFT is performed (optional)
-/// \param n [in] Length of the transformed axis of the output (optional)
+/// \param norm [in] How the normalization is applied (default, backward)
+/// \param axis [in] Axis over which FFT is performed (default, -1)
```

Comment:
Actually I have one question, why do we make a difference between `norm`/`axis` and `n` ? Only the last parameter is a `std::optional`.

Reply:
Basically, two reasons.
1. Firstly, this parameter may not be frequently used. So, I would like to set `None` as a default parameter.
2. Secondly, we cannot give a default value in compile time (cannot hard-code), which is affected by the in/out view extents. In contrast, `norm` and `axis` can have fixed values.

Reply:
> norm and axis can have fixed values.

We could uniformly use `std::optional` ? In 2D for example,

```cpp
template <typename ExecutionSpace, typename InViewType, typename OutViewType>
void fft2(const ExecutionSpace& exec_space, const InViewType& in,
          OutViewType& out,
          std::optional<KokkosFFT::Normalization> norm = std::nullopt,
          std::optional<axis_type<2>> axes = std::nullopt, std::optional<shape_type<2>> s = std::nullopt)
```

This would allow to specify only one parameter such as

```cpp
fft(exec_space, in, out, std:::nullopt, {1, 0})
```

Reply:
We can also do that.
The current implementation helps doxygen to auto generate default parameters, for example,

```C++
template<typename ExecutionSpace, typename InViewType, typename OutViewType>
void KokkosFFT::fft(
  const ExecutionSpace &exec_space,
  const &in,
  OutViewType &out,
  KokkosFFT::Normalization norm = KokkosFFT::Normalization::backward,
  int axis = -1,
  std::optional<std::size_t> n = std::nullopt)
```

Reply:
Closing, this discussion is not critical for this PR.